gooddata-sdk/gooddata_sdk/table.py

```diff
@@ -193,12 +215,255 @@ def _as_table(response: ExecutionResponse) -> ExecutionTable:
+    order: int
+    total: TotalDefinition
+
+    @classmethod
+    def create(cls, total: TotalDefinition, measures: list[InsightMetric]) -> TotalDefinitionOrdering:
+        metric_ordering = [m.local_id for m in measures]
+        return cls(
+            function_order=_TOTAL_ORDER.index(total.aggregation.upper()),
+            order=metric_ordering.index(total.metric_local_id),
+            total=total,
+        )
+
+
+def _create_dimension(bucket: InsightBucket, measures_item_identifier: Optional[str] = None) -> TableDimension:
+    item_ids = [a.local_id for a in bucket.attributes]
+    if measures_item_identifier is not None:
+        item_ids.append(measures_item_identifier)
+    return TableDimension(
+        item_ids=item_ids,
+        idx=_GET_DIM_INDEX_OF_BUCKET_TYPE[bucket.type],
+        sorting=None,
+    )
+
+
+def _create_dimensions(insight: Insight) -> list[TableDimension]:
+    # TODO: measures item id placement should reflect table transposition setting
+    measures_item_identifier = _MEASURE_GROUP_IDENTIFIER if insight.metrics else None
+    row_bucket = insight.get_bucket_of_type(BucketType.ROWS)
+    col_bucket = insight.get_bucket_of_type(BucketType.COLS)
+    return [
+        _create_dimension(row_bucket),
+        _create_dimension(col_bucket, measures_item_identifier),
+    ]
+
+
+def _marginal_total_local_identifier(total: InsightTotal, dim_idx: int) -> str:
+    return f"marginal_total_{total.type}_{total.measure_id}_by_{total.attribute_id}_{dim_idx}"
+
+
+def _sub_total_column_local_identifier(total: InsightTotal, dim_idx: int) -> str:
+    return f"subtotal_column_{total.type}_{total.measure_id}_by_{total.attribute_id}_{dim_idx}"
+
+
+def _sub_total_row_local_identifier(total: InsightTotal, dim_idx: int) -> str:
+    return f"subtotal_row_{total.type}_{total.measure_id}_by_{total.attribute_id}_{dim_idx}"
+
+
+def _grand_total_local_identifier(total: InsightTotal, dim_idx: int) -> str:
+    return f"total_of_totals_{total.type}_{total.measure_id}_by_{total.attribute_id}_{dim_idx}"
+
+
+def _total_local_identifier(total: InsightTotal, dim_idx: int) -> str:
+    return f"total_{total.type}_{total.measure_id}_by_{total.attribute_id}_{dim_idx}"
+
+
+def _convert_total_dimensions(
+    total: InsightTotal, dimension: TableDimension, idx: int, all_dims: list[TableDimension]
+) -> Any:
+    item_idx = dimension.item_ids.index(total.attribute_id)
+    total_dim_items = dimension.item_ids[0:item_idx]
+    if _MEASURE_GROUP_IDENTIFIER in dimension.item_ids and _MEASURE_GROUP_IDENTIFIER not in total_dim_items:
+        total_dim_items.append(_MEASURE_GROUP_IDENTIFIER)
+    total_dims = [TotalDimension(idx=d.idx, items=d.item_ids) for d in all_dims if d.idx != idx]
+    if len(total_dim_items) > 0:
+        total_dims.append(TotalDimension(idx=idx, items=total_dim_items))
+    return total_dims
+
+
+def _get_additional_totals(insight: Insight, dimensions: list[TableDimension]) -> list[TotalDefinition]:
```

Comment:
This function is massive. Could we make it smaller? Split it to smaller functions? Utilize dataclass methods?